src/api/cardsApi.test.js

```diff
@@ -1,42 +1,151 @@
+    it('throws an error when text doesn\'t exist', (done) => {
+      expect.assertions(1);
+      saveCard({ columnId: 1 }).catch((err) => {
+        expect(err.message).toEqual('Invalid card request');
+        done();
+      });
+    });
+
+    it('throws an error when text is empty', (done) => {
+      expect.assertions(1);
+      saveCard({ text: '', columnId: 1 }).catch((err) => {
+        expect(err.message).toEqual('Invalid card request');
+        done();
+      });
+    });
+
+    it('throws an error when text is only spaces', (done) => {
+      expect.assertions(1);
+      saveCard({ text: '  ', columnId: 1 }).catch((err) => {
+        expect(err.message).toEqual('Invalid card request');
+        done();
+      });
+    });
+
+    it('throws an error when columnId doesn\'t exist', (done) => {
+      expect.assertions(1);
+      saveCard({ text: 'hello' }).catch((err) => {
+        expect(err.message).toEqual('Invalid card request');
+        done();
+      });
+    });
+
+    it('throws an error when columnId null', (done) => {
+      expect.assertions(1);
+      saveCard({ text: 'hello', columnId: null }).catch((err) => {
+        expect(err.message).toEqual('Invalid card request');
+        done();
+      });
+    });
+  });
+
+  describe('successful request', () => {
+    it('sends a saved card with an id', async () => {
+      const card = {
+        columnId: 1, text: 'text', username: 'John Doe', voters: []
+      };
+      await saveCard(card);
+      expect(fetchApiSpy).toHaveBeenCalledWith({
+        endpoint: '/cards',
+        method: 'POST',
+        body: card
+      });
+    });
   });
 });
 
-it('throws an error when text is only spaces', (done) => {
-  expect.assertions(1);
-  saveCard({ text: '  ', columnId: 1 }).catch((err) => {
-    expect(err.message).toEqual('Invalid card request');
-    done();
+describe('editCard', () => {
+  describe('invalid request', () => {
+    it('throws an error when text is empty', (done) => {
+      expect.assertions(1);
+      editCard(1, '').catch((err) => {
+        expect(err.message).toEqual('Invalid card request');
+        done();
+      });
+    });
+
+    it('throws an error when text is only spaces', (done) => {
+      expect.assertions(1);
+      editCard(1, '   ').catch((err) => {
+        expect(err.message).toEqual('Invalid card request');
+        done();
+      });
+    });
+
+    it('throws an error when text null', (done) => {
+      expect.assertions(1);
+      editCard(1, null).catch((err) => {
+        expect(err.message).toEqual('Invalid card request');
+        done();
+      });
+    });
+  });
+
+  describe('valid request', () => {
+    it('calls fetchWrapper with updated text', async () => {
+      fetchApiSpy = jest.spyOn(fetchApi, 'fetchWrapper').mockImplementationOnce(() => {});
+      const newText = 'text';
+      await editCard(1, newText);
+      expect(fetchApiSpy).toHaveBeenCalledWith({
+        endpoint: '/cards/1',
```

Comment:
please change URL to not be hard coded cardID, then we can extract /cards to constant for all test cases in file